### PR TITLE
Added Azure Active Directory Support

### LINKED ISSUE #29: Add support for Microsoft Azure Active Directory Apps
Currently there are two node libraries that support azure active directory. Would be great to see an example how the passport version could be supported here

https://github.com/AzureAD/passport-azure-ad

As an alternative there's a core ADAL node library as well.

https://github.com/AzureAD/azure-activedirectory-library-for-nodejs

### PR DESCRIPTION
**Pull request by Nop0x**
### Description
Added a working template for authentication with Azure AD.

#### Related issues
- fixes #29 

### Checklist

- [X] New tests added or existing tests modified to cover all changes
- [X] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by dhmlau** [comment]
@Nop0x , thanks for the PR.  Could you please sign the CLA? 
https://cla.strongloop.com/agreements/strongloop/loopback-example-passport

@raymondfeng, could you please review?  thanks.

**Comment by Nop0x** [comment]
@dhmlau signed and accepted!

**Comment by dhmlau** [comment]
Thanks @Nop0x.  Your PR has landed!

File: providers.json.template [hunk 1]
```diff
@@ -101,5 +101,27 @@
     "scope": ["email", "profile"],
     "link": true,
     "failureFlash": true
+  },
+    "azure-ad": {
```
Comment (blackbeario) [inline providers.json.template]: Indentation is incorrect. Should be 2 spaces, not 4.

**Comment by blackbeario** [comment]
Would be nice to have an example of Azure AD with BearerStrategy. It seems impossible to find online.

**Comment by Pieter-Dirix** [comment]
The provided example doesn't work. Please update the docs and include Azure AD

**Pull request merged**